backend/app/utils/Ocr.scala

```diff
@@ -78,20 +78,45 @@ object Ocr extends Logging {
     val tempFile = tmpDir.resolve(s"${inputFilePath.getFileName}.ocr.pdf")
     val stdout = mutable.Buffer.empty[String]
 
-    def process(flag: OcrMyPdfFlag): Int = {
-      val cmd = s"ocrmypdf ${flag.flag} -l $lang ${dpi.map(dpi => s"--image-dpi $dpi").getOrElse("")} ${inputFilePath.toAbsolutePath} ${tempFile.toAbsolutePath}"
+    def process(flag: OcrMyPdfFlag, overrideFile: Option[Path] = None): Int = {
+      val sourceFilePath = overrideFile.getOrElse(inputFilePath)
+      val cmd = s"ocrmypdf ${flag.flag} -l $lang ${dpi.map(dpi => s"--image-dpi $dpi").getOrElse("")} ${sourceFilePath.toAbsolutePath} ${tempFile.toAbsolutePath}"
       val process = Process(cmd, cwd = None, extraEnv = "TMPDIR" -> tmpDir.toAbsolutePath.toString)
       process.!(ProcessLogger(stdout.append(_), stderr.append))
     }
 
+    def decryptWithQpdf(decryptTempFile: Path): Boolean = {
+      val cmd = s"qpdf --decrypt ${inputFilePath.toAbsolutePath} ${decryptTempFile.toAbsolutePath}"
+      val process = Process(cmd, cwd = None)
+      val qpdfExitCode = process.!(ProcessLogger(stdout.append(_), stderr.append))
+      if (qpdfExitCode != 0) {
+        logger.info(s"Failed to decrypt with qpdf (exit code ${qpdfExitCode} - file is likely encrypted with a user password.")
+      }
+      qpdfExitCode == 0
+    }
+
     val redoOcrExitCode = process(RedoOcr)
     val exitCode = if (redoOcrExitCode == 2) {
       // Exit code 2 from ocrmypdf is an input file error, we've noticed that this can be an error with --redo-ocr, and that
       // running with --skip-text instead results in success. For example, if a PDF has a user fillable form then it can't
       // be ocrd with --redo-ocr set. See https://github.com/guardian/giant/pull/68 for details of --skip-text vs --redo-ocr
       logger.info(s"Got input file error from ocrmypdf with --redo-ocr for ${inputFilePath.getFileName}, attempting with --skip-text")
       process(SkipText)
-    } else redoOcrExitCode
+    } else if (redoOcrExitCode == 8) {
```

Comment:
Thanks Zeke this is definitely an edge case I hadn't thought about, I'll refactor a bit so that the next person to add an extra step has less chance of making the same mistake